### PR TITLE
Implement snapshot records for last_success and last_failure on SnapshotLifecyclePolicyMetadata.

### PR DESCRIPTION
**Pull request by codebrain**
Relates to: https://github.com/elastic/elasticsearch/pull/45245

https://github.com/elastic/elasticsearch/blob/7.4/client/rest-high-level/src/main/java/org/elasticsearch/client/slm/SnapshotLifecyclePolicyMetadata.java

### CODE REVIEW

File: src/Nest/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs [hunk 1]
```diff
@@ -52,6 +52,31 @@ public class SnapshotLifecyclePolicyMetadata
...
+		public SnapshotInvocationRecord LastSuccess { get; set; }
+
+		/// <summary>
+		///	 Information about the last time the policy failed to initiate a snapshot
+		/// </summary>
+		[DataMember(Name = "last_failure")]
+		public SnapshotInvocationRecord LastFailure { get; set; }
+	}
+
+	public class SnapshotInvocationRecord
```
Comment (russcam) [inline src/Nest/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs]: rename this to `SnapshotLifecycleInvocationRecord` in line with other SLM type names? The Java source uses `SnapshotInvocationRecord`, but has namespaces to reuse type names across the codebase.

File: src/Nest/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs [hunk 2]
```diff
@@ -52,6 +52,31 @@ public class SnapshotLifecyclePolicyMetadata
...
+	{
+		[DataMember(Name = "snapshot_name")]
+		public string SnapshotName { get; set; }
+
+		[DataMember(Name = "time_string")]
+		public string TimeString { get; set; }
```
Comment (russcam) [inline src/Nest/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs]: Not sure we need to expose this, given `Time` will deserialize to a `DateTimeOffset`?

Comment (Mpdreamz) [reply src/Nest/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs]: ++ to omitting this.

**Comment by Mpdreamz** [comment]
Usual suspects failed on integration tests for tickets have already been opened.

**Comment by codebrain** [comment]
Ported to `master`

**Pull request merged**